oligo_designer_toolsuite/database/_oligo_database.py

```diff
@@ -701,33 +715,153 @@ def write_oligosets_to_yaml(
         with open(file_yaml, "w") as handle:
             yaml.dump(yaml_dict, handle, Dumper=CustomYamlDumper, default_flow_style=False, sort_keys=False)
 
-        return file_yaml
-
     def write_oligosets_to_table(
-        self, dir_output: str = "oligosets", region_ids: Union[str, List[str]] = None
-    ) -> str:
+        self,
+        properties: str | list[str],
+        top_n_sets: int,
+        ascending: bool,
+        filename: str = "oligosets",
+        dir_output: str | None = None,
+        region_ids: str | list[str] | None = None,
+    ) -> None:
+        # Check formatting
+        properties = check_if_list(properties)
+        region_ids = check_if_list(region_ids) if region_ids else self.database.keys()
+
+        csv_table = list()
+
+        for region_id in region_ids:
+            oligosets_region = self.oligosets[region_id]
+            oligosets_oligo_columns = [col for col in oligosets_region.columns if col.startswith("oligo_")]
+            oligosets_score_columns = [
+                col for col in oligosets_region.columns if col.startswith("set_score_")
+            ]
+
+            oligosets_region.sort_values(by=oligosets_score_columns, ascending=ascending, inplace=True)
+            oligosets_region = oligosets_region.head(top_n_sets)[oligosets_oligo_columns]
+            oligosets_region.reset_index(inplace=True, drop=True)
+
+            # iterate through all oligo sets
+            for oligoset_idx, oligoset in oligosets_region.iterrows():
+                oligoset_id = f"oligoset_{oligoset_idx + 1}"
+                for oligo_id in oligoset:
+                    entry = {
+                        "region_id": region_id,
+                        "oligoset_id": oligoset_id,
+                        "oligo_id": oligo_id,
+                    }
+
+                    for property in properties:
+                        if property in self.database[region_id][oligo_id]:
+                            oligo_property = self.database[region_id][oligo_id][property]
+                            # format oligo properties: flatten lists of lists, join string lists with comma, keep strings as-is, None -> empty list
+                            if oligo_property:
+                                if (
+                                    sum(len(sublist) for sublist in check_if_list_of_lists(oligo_property))
+                                    == 1
+                                ):
+                                    oligo_property = flatten_property_list(oligo_property)
+                                    if len(oligo_property) == 1:
+                                        oligo_property = oligo_property[0]
```

Comment:
sorry, don't know the exact format of the oligo properties; do we need any else condition if the properties don't behave as expected that we can write them?